src/blockdev.rs

```diff
@@ -461,6 +464,205 @@ impl Mount {
     }
 }
 
+#[derive(Debug)]
+pub struct SavedPartitions {
+    sector_size: Option<u64>,
```

Comment:
Nit: I think an `Option<NonZeroU16>` could be enough here. But most of the logic below really expects a `None` to never happen here; it would be good to rework the `new` flow in order to be able to drop the `Option` here.

Reply:
I used `u64` because that's the type of `gpt.sector_size`, of which this is a copy.

`sector_size` is `None` in the case where we return early from `new()` (because we have no filters) and haven't looked at the GPT at all.  I think that's a useful check but I agree that the `expect`/`unwrap` are ugly; I'll unwrap `sector_size` early in `write()` and store it as a local.